### PR TITLE
Updated MainResourceController to only purge a Resource if purge para…

### PR DESCRIPTION
**Pull request by samuelmale**
…m is equal to true

Ticket: https://issues.openmrs.org/browse/RESTWS-703

### CODE REVIEW

**Comment by samuelmale** [comment]
Could you please look at this! I'm not sure why we have the above issues with the POM file yet I didn't even touch it.

@wluyima , @dkayiwa

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java [hunk 1]
```diff
@@ -155,11 +154,19 @@ public Object delete(@PathVariable("resource") String resource, @PathVariable("u
 	@RequestMapping(value = "/{resource}/{uuid}", method = RequestMethod.DELETE, params = "purge")
```
Comment (wluyima) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: I believe changing the value of this **params** attribute is enough without touching the method code

Comment (wluyima) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: That way you don't have to bake the void/retire logic in the method.

Comment (samuelmale) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: I think I'm missing something @wluyima . 

>A resource should only be purged when the purge parameter is set to true

The above being the title of this ticket, how would this be possible without changing the method code accordingly. I think I'm not understanding you ..

Comment (wluyima) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: I think the params attribute can take a spring EL expression, I think in this case you need to set the value for the purge parameter to true, you might want to look at the spring docs for the params attribute on RequestMapping for the exact syntax

Comment (wluyima) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: That way spring will only invoke this controller if purge is set explicitly set to true by the client, you won't have to make any code changes in the controller method.

**Comment by samuelmale** [comment]
Perfect, resolved all issues @wluyima

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java [hunk 2]
```diff
@@ -152,7 +152,7 @@ public Object delete(@PathVariable("resource") String resource, @PathVariable("u
 	 * @param response
 	 * @throws Exception
 	 */
-	@RequestMapping(value = "/{resource}/{uuid}", method = RequestMethod.DELETE, params = "purge")
+	@RequestMapping(value = "/{resource}/{uuid}", method = RequestMethod.DELETE, params = "purge=true")
```
Comment (wluyima) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java]: There you go! See how you didn't have to make any changes within the controller method? And everything still works

**Pull request merged**